### PR TITLE
Rename VHLO_DenseIntOrFPElementsAttr to VHLO_TensorAttr

### PR DESCRIPTION
**Pull request by burmako**
This is one of the several steps towards aligning VhloAttrs.td with the StableHLO spec.

Keeping the forked name is understandable, but I think that alignment with the spec is more important. See #1047.

### CODE REVIEW

File: stablehlo/dialect/VhloBytecode.cpp [hunk 1]
```diff
@@ -220,6 +214,12 @@ enum AttributeCode {
   ///   }
   kStringAttr = 22,
 
+  ///   TensorAttr {
+  ///     type: Type
+  ///     data: blob
+  ///   }
+  kTensorAttr = 17,
```
Comment (GleasonK) [inline stablehlo/dialect/VhloBytecode.cpp:221]: Merge the other bytecode modifications then lets renumber since we can break still. Could open an issue to follow up.

File: stablehlo/dialect/VhloBytecode.cpp [hunk 2]
```diff
@@ -597,13 +595,13 @@ LogicalResult VhloBytecodeInterface::writeAttribute(
...
             return success();
           })
-      .Case<ArrayV1Attr, DenseIntOrFPElementsV1Attr, DictionaryV1Attr,
-            FlatSymbolRefV1Attr, FloatV1Attr, IntegerV1Attr, StringV1Attr,
-            TypeV1Attr>([&](auto attr) {
-        LOG_WRITE_CALL;  // Forked attrs
-        write(attr, writer);
-        return success();
-      })
+      .Case<ArrayV1Attr, DictionaryV1Attr, FlatSymbolRefV1Attr, FloatV1Attr,
```
Comment (GleasonK) [inline stablehlo/dialect/VhloBytecode.cpp:598]: This split for `.Case` above and `.Case` here is that some are upstream and some are StableHLO. We can merge these two case statements since this distinction no longer matters (maybe this was done in a different PR)

**Review by GleasonK — APPROVED** [review]

File: stablehlo/dialect/VhloOps.cpp [hunk 3]
```diff
@@ -178,23 +178,23 @@ ParseResult parseFunctionBody(OpAsmParser& parser, Attribute& name,
   return success();
 }
 
-void DenseIntOrFPElementsV1Attr::print(mlir::AsmPrinter& p) const {
```
Comment (GleasonK) [inline stablehlo/dialect/VhloOps.cpp:181]: This file could use some sorting eventually, but nothing blocking.

File: stablehlo/dialect/VhloBytecode.cpp (hunk 1 above)
Comment (burmako) [reply stablehlo/dialect/VhloBytecode.cpp:221]: Yes, this is done in the final PR of this chain ("Audit StablehloAttrs.td").

File: stablehlo/dialect/VhloBytecode.cpp (hunk 2 above)
Comment (burmako) [reply stablehlo/dialect/VhloBytecode.cpp:598]: Yeah, same here :)

File: stablehlo/dialect/VhloOps.cpp (hunk 3 above)
Comment (burmako) [reply stablehlo/dialect/VhloOps.cpp:181]: Agreed! I was planning to finish externalizing as much of VhloOps.cpp as possible first (all verifiers and shape functions have been moved, then I wanted to move reifyReturnTypeShapes and prettyprinters/parsers), and then alphabetically sort everything.
